test/native/integration-test-helpers/add-block.js

```diff
@@ -44,7 +44,19 @@ export const addBlock = async (
 		},
 	} );
 
-	fireEvent.press( await waitFor( () => getByText( blockName ) ) );
+	const blockButton = await waitFor( () => getByText( blockName ) );
+	// Blocks can perform belated state updates after they are inserted.
+	// To avoid potential `act` warnings, we ensure that all timers and queued
+	// microtasks are executed.
+	await withFakeTimers( async () => {
+		fireEvent.press( blockButton );
+		// Run all timers, in case any performs a state updates.
+		// Column block example: https://t.ly/NjTs
+		act( () => jest.runOnlyPendingTimers() );
```

Comment:
I haven't checked if other blocks use timers to update their local state when created. I agree that, if this is only needed for a specific block, we could execute in the associated test files instead of in a generic way.

My purpose with this approach was to simplify the block insertion, as I feel that it won't be very common to assert potential state updates that happen during the insertion. I assumed that all of them acted as part of the `addBlock` helper. However, in case we need to, we could run the timers conditionally via a configuration parameter.

I'll check if this is needed in more blocks, and depending on that, we could decide if we rather run the timers generally here or in each test 👍 .

Reply:
Your expressed purpose makes a lot of sense. I recognize that our testing environment needs to find a balance between convenience and comprehensibility. 

To be clear, I am not opposed to inserting this global timer into the `addBlock` helper. I merely wanted to note its potential for breeding confusion due to the obscure nature of this timers run. The inline comments will likely help to mitigate that, provided the test author reads the source of `addBlock` when using it. 

Please feel free to leave this implementation as-is. A configuration parameter could arguably be unnecessary complexity.